### PR TITLE
Add Instances.mapK

### PR DESCRIPTION
**Pull request by joroKr21**

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -86,7 +86,9 @@ lazy val deriving = crossProject(JSPlatform, JVMPlatform)
   .settings(mimaSettings)
   .settings(
      mimaBinaryIssueFilters ++= Seq(
-       ProblemFilters.exclude[ReversedMissingMethodProblem]("shapeless3.deriving.internals.ErasedProductInstances.erasedProject")
+       ProblemFilters.exclude[ReversedMissingMethodProblem]("shapeless3.deriving.internals.ErasedInstances.erasedMapK"),
+       ProblemFilters.exclude[ReversedMissingMethodProblem]("shapeless3.deriving.internals.ErasedProductInstances.erasedProject"),
+       ProblemFilters.exclude[ReversedMissingMethodProblem]("shapeless3.deriving.internals.ErasedProductInstances.erasedMapK")
```
Comment (milessabin) [inline build.sbt:91]: Don't we need an exclusion for `shapeless3.deriving.internals.ErasedCoproductInstances.erasedMapK` as well?

**Review by milessabin — APPROVED** [review]
Aside from the MiMa exclusions this LGTM!

File: build.sbt (hunk 1 above)
Comment (joroKr21) [reply build.sbt:91]: No, because it's final. I guess if we made `Instances` and `ProductInstances` sealed we would need no exclusions at all.

Comment (milessabin) [reply build.sbt:91]: I'm fine with making it sealed, but shouldn't [this](https://github.com/lightbend/mima/pull/583) mean none of these should be necessary?

Comment (joroKr21) [reply build.sbt:91]: Not for Scala 3: https://github.com/lightbend/mima/pull/583#issuecomment-734965304

Comment (milessabin) [reply build.sbt:91]: Ahh ... right. Then this is good to go :smile:

**Pull request merged**